src/core/thread/mesh_forwarder.cpp

```diff
@@ -652,6 +653,21 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
                 VerifyOrExit(mMle.IsRouterIdValid(mMle.GetRouterId(rloc16)), error = kThreadError_Drop);
                 mMeshDest = rloc16;
             }
+            else if (mMle.IsAnycastLocator(ip6Header.GetDestination()))
+            {
+                // only support Leader ALOC for now
+                aloc16 = HostSwap16(ip6Header.GetDestination().mFields.m16[7]);
+
+                if (aloc16 == 0xfc00)
+                {
+                    mMeshDest = mMle.GetRloc16(mMle.GetLeaderId());
+                }
+                else
+                {
```

Comment:
Till my understanding, the difference here is on-mesh and external, but not the roles of the node. If we only support on-mesh so far, we do not need to use if/else here?